### PR TITLE
test: add e2e test cases for route with plugin `proxy-rewrite`

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [x] E2E test

- Related issues

#636

### CODE REVIEW

File: api/test/e2e/route_with_other_plugin_test.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+ * limitations under the License.
+ */
+package e2e
+
+import (
+	"net/http"
+	"testing"
+)
+
+func TestRoute_With_Other_Plugin(t *testing.T) {
```
Comment (membphis) [inline api/test/e2e/route_with_other_plugin_test.go:24]: bad name, mainly about `proxy-rewrite` plugin

Comment (membphis) [reply api/test/e2e/route_with_other_plugin_test.go:24]: and please update the file name too

File: api/test/e2e/route_with_other_plugin_test.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	tests := []HttpTestCase{
+		{
+			caseDesc:     "make sure the route is not created ",
```
Comment (membphis) [inline api/test/e2e/route_with_other_plugin_test.go:27]: remove the last space

File: api/test/e2e/route_with_other_plugin_test.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+			caseDesc:     "verify route that rewrite host and uri",
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "uri: /plugin_proxy_rewrite\nhost: test.com",
+			Sleep:        sleepTime,
+		},
+		{
+			caseDesc: "create route that will rewrite headers",
```
Comment (membphis) [inline api/test/e2e/route_with_other_plugin_test.go:69]: should be `update`, all right?

File: api/test/e2e/route_with_other_plugin_test.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+			Object:       APISIXExpect(t),
+			Method:       http.MethodGet,
+			Path:         "/hello",
+			Headers:      map[string]string{"X-Api-Version": "v1"},
+			ExpectStatus: http.StatusOK,
+			ExpectBody:   "x-api-version: v2",
+			Sleep:        sleepTime,
+		},
+		{
+			caseDesc: "create route that using regex_uri",
```
Comment (membphis) [inline api/test/e2e/route_with_other_plugin_test.go:106]: ditto

**Review by membphis — CHANGES_REQUESTED** [review]

File: api/test/e2e/route_with_other_plugin_test.go (hunk 1 above)
Comment (nic-chen) [reply api/test/e2e/route_with_other_plugin_test.go:24]: fixed.

File: api/test/e2e/route_with_other_plugin_test.go (hunk 2 above)
Comment (nic-chen) [reply api/test/e2e/route_with_other_plugin_test.go:27]: fixed.

File: api/test/e2e/route_with_other_plugin_test.go (hunk 3 above)
Comment (nic-chen) [reply api/test/e2e/route_with_other_plugin_test.go:69]: fixed.

File: api/test/e2e/route_with_other_plugin_test.go (hunk 4 above)
Comment (nic-chen) [reply api/test/e2e/route_with_other_plugin_test.go:106]: fixed.

File: api/test/e2e/route_with_plugin_proxy_rewrite_test.go [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+		{
+			caseDesc: "update route using regex_uri",
+			Object:   MangerApiExpect(t),
+			Method:   http.MethodPut,
+			Path:     "/apisix/admin/routes/r1",
+			Body: `{
+				"uri": "/test/*",
+				"plugins": {
+					"proxy-rewrite": {
+					"regex_uri": ["^/test/(.*)/(.*)/(.*)", "/$1_$2_$3"]
```
Comment (gxthrj) [inline api/test/e2e/route_with_plugin_proxy_rewrite_test.go:114]: Need to be indented

**Review by membphis — APPROVED** [review]
LGTM, nice

File: api/test/e2e/route_with_plugin_proxy_rewrite_test.go (hunk 5 above)
Comment (nic-chen) [reply api/test/e2e/route_with_plugin_proxy_rewrite_test.go]: fixed.

**Review by gxthrj — APPROVED** [review]

**Pull request merged**